pkg/operator/staticpod/controller/node/node_controller.go

```diff
@@ -72,9 +77,26 @@ func (c NodeController) sync(ctx context.Context, syncCtx factory.SyncContext) e
 			}
 		}
 		if found {
-			newTargetNodeStates = append(newTargetNodeStates, originalOperatorStatus.NodeStatuses[i])
+			newTargetNodeState := applyoperatorv1.NodeStatus().
+				WithNodeName(originalOperatorStatus.NodeStatuses[i].NodeName).
+				WithCurrentRevision(originalOperatorStatus.NodeStatuses[i].CurrentRevision).
+				WithTargetRevision(originalOperatorStatus.NodeStatuses[i].TargetRevision).
+				WithLastFailedRevision(originalOperatorStatus.NodeStatuses[i].LastFailedRevision).
+				WithLastFailedReason(originalOperatorStatus.NodeStatuses[i].LastFailedReason).
+				WithLastFailedCount(originalOperatorStatus.NodeStatuses[i].LastFailedCount).
+				WithLastFallbackCount(originalOperatorStatus.NodeStatuses[i].LastFallbackCount).
+				WithLastFailedRevisionErrors(originalOperatorStatus.NodeStatuses[i].LastFailedRevisionErrors...)
+			if originalOperatorStatus.NodeStatuses[i].LastFailedTime != nil {
+				newTargetNodeState.WithLastFailedTime(*originalOperatorStatus.NodeStatuses[i].LastFailedTime)
+			}
+			newTargetNodeStates = append(newTargetNodeStates, newTargetNodeState)
 		} else {
 			syncCtx.Recorder().Warningf("MasterNodeRemoved", "Observed removal of master node %s", nodeState.NodeName)
+			removeAtIndex := i
+			if i > 0 {
+				removeAtIndex = removeAtIndex - 1
+			}
+			jsonPatch.WithRemove(fmt.Sprintf("/status/nodeStatuses/%d", removeAtIndex), jsonpatch.NewTestCondition(fmt.Sprintf("/status/nodeStatuses/%d/nodeName", removeAtIndex), nodeState.NodeName))
```

Comment:
I'm not sure I understand this subtraction. Could you elaborate why you need to do this?

I'm confused because whether `i==0` or `i==1` we end up removing the same index?

Reply:
added a comment, ptal

Reply:
Looking much better, thanks!